central/complianceoperator/v2/profiles/datastore/datastore_impl.go

```diff
@@ -3,129 +3,98 @@ package datastore
-	} else if !ok {
-		return nil, sac.ErrResourceAccessDenied
-	}
-
 	return d.store.GetByQuery(ctx, query)
 }
 
 // UpsertProfile adds the profile to the database
-func (d *datastoreImpl) UpsertProfile(ctx context.Context, profile *storage.ComplianceOperatorProfileV2, clusterID string, profileUID string) error {
-	if ok, err := complianceSAC.WriteAllowed(ctx); err != nil {
-		return err
-	} else if !ok {
+func (d *datastoreImpl) UpsertProfile(ctx context.Context, profile *storage.ComplianceOperatorProfileV2) error {
+	if !complianceSAC.ScopeChecker(ctx, storage.Access_READ_WRITE_ACCESS).IsAllowed(sac.ClusterScopeKey(profile.GetClusterId())) {
 		return sac.ErrResourceAccessDenied
 	}
 
-	tx, err := d.db.Begin(ctx)
-	if err != nil {
-		return err
-	}
-	ctx = pgPkg.ContextWithTx(ctx, tx)
-
-	if err := d.store.Upsert(ctx, profile); err != nil {
-		return wrapRollback(ctx, tx, errors.Wrapf(err, "error adding profile %s", profile.GetProfileId()))
-	}
-
-	profileEdge := &storage.ComplianceOperatorProfileClusterEdge{
-		Id:         uuid.NewV4().String(),
-		ProfileId:  profile.GetId(),
-		ProfileUid: profileUID,
-		ClusterId:  clusterID,
-	}
-
-	err = d.profileEdgeStore.Upsert(ctx, profileEdge)
-	if err != nil {
-		return wrapRollback(ctx, tx, errors.Wrapf(err, "error adding profile for cluster %s", clusterID))
-	}
-
-	err = tx.Commit(ctx)
-	if err != nil {
-		return wrapRollback(ctx, tx, errors.Wrapf(err, "error adding profile for cluster %s", clusterID))
-	}
-	return nil
+	return d.store.Upsert(ctx, profile)
 }
 
 // DeleteProfileForCluster removes a profile from the database
 func (d *datastoreImpl) DeleteProfileForCluster(ctx context.Context, uid string, clusterID string) error {
-	if ok, err := complianceSAC.WriteAllowed(ctx); err != nil {
-		return err
-	} else if !ok {
+	if !complianceSAC.ScopeChecker(ctx, storage.Access_READ_WRITE_ACCESS).IsAllowed(sac.ClusterScopeKey(clusterID)) {
 		return sac.ErrResourceAccessDenied
 	}
 
-	_, storeErr := d.profileEdgeStore.DeleteByQuery(ctx, search.NewQueryBuilder().
+	_, err := d.store.DeleteByQuery(ctx, search.NewQueryBuilder().
 		AddExactMatches(search.ClusterID, clusterID).
-		AddExactMatches(search.ComplianceOperatorProfileUID, uid).ProtoQuery())
-	return storeErr
+		AddDocIDs(uid).ProtoQuery())
+	return err
+}
+
+// GetProfilesByCluster gets the list of profiles for a given cluster
+func (d *datastoreImpl) GetProfilesByCluster(ctx context.Context, clusterID string) ([]*storage.ComplianceOperatorProfileV2, error) {
+	return d.store.GetByQuery(ctx, search.NewQueryBuilder().AddExactMatches(search.ClusterID, clusterID).ProtoQuery())
 }
 
-// GetProfileEdgesByCluster gets the list of profile edges for a given cluster
-func (d *datastoreImpl) GetProfileEdgesByCluster(ctx context.Context, clusterID string) ([]*storage.ComplianceOperatorProfileClusterEdge, error) {
-	if ok, err := complianceSAC.ReadAllowed(ctx); err != nil {
+// GetProfileSummaryForClusters gets the list of profiles for a given clusters
```

Comment:
nit: We could update this to say this returns a profile summary, and not the complete profile reference.